rstudio/rstudio
# Include build info in Electron app

### Intent

Addresses #10555. Allows Electron app to report build information (version, commit-hash, build date, etc.) on session launch failure page.

Also tweaked `make-package` for Mac to report usage info when given an unknown argument (e.g. `./make-package --help`), and to avoid invoking the cleanup stuff twice.

### Approach

Use cmake configure_file to embed the fields. For dev workflow, uses some canned values; for a make-package build, will contain the real info.

### Automated Tests

None

### QA Notes

Launch RStudio.app from terminal with RSTUDIO_SESSION_EXIT_ON_STARTUP environment variable set to force display of the "Error starting R" page, and confirm that the version info shown matches the build.

`RSTUDIO_SESSION_EXIT_ON_STARTUP=1 ./RStudio.app/Contents/MacOS/RStudio`

Also try launching with --version and confirm that correct value is shown:

`./RStudio.app/Contents/MacOS/RStudio --version`

### Checklist
